packages/x-charts/src/internals/plugins/corePlugins/useChartSeries/processSeries.ts

```diff
@@ -73,3 +78,37 @@ export const applySeriesProcessors = <TSeriesType extends ChartSeriesType>(
 
   return processedSeries;
 };
+
+/**
+ * Applies series processors with drawing area to series if defined.
+ * @param processedSeries The processed series groups
+ * @param seriesConfig The series configuration
+ * @param drawingArea The drawing area
+ * @returns Processed series with all transformations applied
+ */
+export const applySeriesProcessors = <TSeriesType extends ChartSeriesType>(
+  processedSeries: ProcessedSeriesWithoutDimensions<TSeriesType>,
+  seriesConfig: ChartSeriesConfig<TSeriesType>,
+  drawingArea: ChartDrawingArea,
+): ProcessedSeries<TSeriesType> => {
+  let processingDetected = false;
+  const modifiedProcessedSeries: ProcessedSeries<TSeriesType> = {};
+
+  // Apply processors on series type per group
+  (Object.keys(processedSeries) as TSeriesType[]).forEach((type) => {
+    const processor = seriesConfig[type]?.seriesProcessorWithDrawingArea;
+    if (processor !== undefined) {
+      const newValue = processor(processedSeries[type] as any, drawingArea);
+
+      if (newValue && newValue !== processedSeries[type]) {
+        processingDetected = true;
+        (modifiedProcessedSeries as any)[type] = newValue;
+      }
+    }
+  });
+
+  if (!processingDetected) {
+    return processedSeries as ProcessedSeries<TSeriesType>;
+  }
```

Comment:
Is `processingDetected` just to avoid breaking selectors' cache when processing results in the same value?